lib/models/asyncapi.js

```diff
@@ -242,24 +243,25 @@ function assignNameToAnonymousMessages(doc) {
   if (doc.hasChannels()) {
     doc.channelNames().forEach(channelName => {
       const channel = doc.channel(channelName);
-      if (channel.hasPublish()) {
-        channel.publish().messages().forEach(m => {
-          if (m.name() === undefined) {
-            m.json()['x-parser-message-name'] = `<anonymous-message-${++anonymousMessageCounter}>`;
-          }
-        });
-      }
-      if (channel.hasSubscribe()) {
-        channel.subscribe().messages().forEach(m => {
-          if (m.name() === undefined) {
-            m.json()['x-parser-message-name'] = `<anonymous-message-${++anonymousMessageCounter}>`;
-          }
-        });
-      }
+      if (channel.hasPublish()) addNameToKey(channel.publish().messages(),++anonymousMessageCounter);
+      if (channel.hasSubscribe()) addNameToKey(channel.subscribe().messages(),++anonymousMessageCounter);
     });
   }
 }
 
+/**
+ * Add anonymous name to key if no name provided.
+ * 
+ * @param {messages} map of messages 
+ */
+function addNameToKey(messages, number) {
```

Comment:
this is a manual change I did, because of cognitive complexity, and code duplication that sonar complained about